src/middlewared/middlewared/plugins/enclosure.py

```diff
@@ -1038,21 +1039,27 @@ def device_slot_set(self, status):
         Returns:
             True if the command succeeded, False otherwise
         """
-        if status == "clear":
-            commands = ["--clear=fault", "--clear=ident"]
-        else:
-            if status == "identify":
-                commands = ["--set=ident"]
-            else:
-                commands = [f"--set={status}"]
-
+        slot = f'{self.slot - 1}'
         ok = True
-        for cmd in commands:
-            cmd = ["sg_ses", "--index", str(self.slot - 1), cmd, f"/dev/{self.enclosure.devname}"]
-            p = subprocess.run(cmd, encoding="utf-8", errors="ignore", stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            if p.returncode != 0:
-                logger.warning("%r failed: %s", cmd, p.stderr)
-                ok = False
+        if status == 'clear':
+            for getit, setit in (('get=fault', 'clear=fault'), ('get=ident', 'clear=ident')):
+                if self.sg3obj.set_control(slot, getit):
+                    # fault or identify light is toggled so try and clear it
+                    self.sg3obj.set_control(slot, setit)
+                    if self.sg3obj.set_control(slot, getit):
+                        # we tried to toggle the light but when we check it again
+                        # it shows as being turned on so at least log a message
+                        # to try and give some idea of what's going on
+                        self.logger.warning('Failed to %s slot %r on %r', status, slot, self.enclosure.devname)
+                        ok = False
+        else:
+            cmds = (('get=ident', 'set=ident'),) if status == 'identify' else ((f'get={status}', f'set={status}'),)
+            for getit, setit in cmds:
+                if self.sg3obj.set_control(slot, getit):
+                    self.sg3obj.set_control(slot, setit)
+                    if self.sg3obj.set_control(slot, getit):
```

Comment:
I am confused by this logic. We expect it to be "off", then set it to "on" and then log an error if it is "on"?